Khan/perseus
# Create PhET simulation widget

## Summary:
Create a new Perseus widget for PhET simulations with the following features:
* Fullscreen mode
* Only allows URLs from the [https://phet.colorado.edu](https://phet.colorado.edu/) origin
  * If the origin is wrong or the resource is not available, we display an error
* Displays simulation in your locale (forgot to put this in the demo so I'll attach a screenshot)
  * If the simulation is not available in your locale, we display a warning banner and load the simulation in English
* Skeleton widget editor
  * This is just to pass a Jest test that expects every widget will have an editor

I have documented the Perseus widget creation process [here](https://khanacademy.atlassian.net/wiki/spaces/LC/pages/3075276818/Creating+a+New+Widget+Type).

Issue: LEMS-2167

<img width="1041" alt="PhET simulation storybook page, contains an iframe with a PhET simulation and a fullscreen button under the bottom right corner of the iframe" src="https://github.com/user-attachments/assets/4c11ab54-6af0-4525-9f0d-f023d1919de6">

<img width="1027" alt="PhET simulation editor storybook page, contains a URL label with an empty input field" src="https://github.com/user-attachments/assets/152015c6-fd20-403f-a7c0-8faecea79440">

## Test plan:

- Verify that the widget shows up appropriately in Storybook
  - Change `this.locale` to some nonexistent locale and verify that the warning shows up
  - Change URL to a non-PhET origin and verify that the error shows up
  - Change URL to a non-existent resource from the PhET origin and verify that the error shows up
  - Click the fullscreen button and verify that the iframe goes into fullscreen mode
 - Verify screen reader accessibility
   - Make sure the fullscreen button, iframe, and iframe contents are all tab-focusable
   - Make sure the fullscreen button, iframe, and iframe contents all have appropriate aria labels
